src/codegen/cfg.rs

```diff
@@ -1308,48 +1367,35 @@ pub fn generate_cfg(
 
     let mut cfg = function_cfg(contract_no, function_no, ns, opt);
 
-    let default_constructor = &ns.default_constructor(contract_no);
-    let func = match function_no {
-        Some(function_no) => &ns.functions[function_no],
-        None => default_constructor,
-    };
-
-    // if the function is a modifier, generate the modifier chain
-    if !func.modifiers.is_empty() {
-        // only function can have modifiers
-        assert_eq!(func.ty, pt::FunctionTy::Function);
-        let public = cfg.public;
-        let nonpayable = cfg.nonpayable;
-
-        cfg.public = false;
+    if let Some(func_no) = function_no {
+        // if the function is a modifier, generate the modifier chain
+        if !ns.functions[func_no].modifiers.is_empty() {
```

Comment:
would the code be slightly easier to read if we had:
```rust
let func = &ns.functions[func_no]:
```
rather than repeating `ns.functions[func_no]` again and again?

Reply:
I've used `func` whenever possible, but after calling `generate_modifier_dispatch` I cannot use it any more because I borrow `ns` as mutable.